File: LintCode/Python3/13.ImplementstrStr().py
# https://www.lintcode.com/problem/implement-strstr/description?_from=ladder&&fromId=1
# Time Complexity O(mn)
# Space Complexity O(1)
class Solution:
    """
    @param source: 
    @param target: 
    @return: return the index
    """

    def strStr(self, source, target):
        l1 = len(source)
        l2 = len(target)

        for i in range(l1 - l2 + 1):
            if source[i:i+l2] == target:
                return i
        return -1

File: lintcode1/Cpp/13.ImplementstrStr().cpp

//https://www.lintcode.com/problem/implement-strstr/description?_from=ladder&&fromId=1
class Solution
{
public:
    /**
     * @param source: 
     * @param target: 
     * @return: return the index
     */
    int strStr(string &source, string &target)
    {
        int lens = source.length(), lent = target.length();
        int i, j;
        if (lent == 0)
        {
            return 0;
        }
        for (i = 0; i <= lens - lent; i++)
        {
            for (j = 0; j < lent; j++)
            {
                if (source[i + j] != target[j])
                {
                    break;
                }
                if (j == lent - 1)
                {
                    return i;
                }
            }
        }
        return -1;
    }
};